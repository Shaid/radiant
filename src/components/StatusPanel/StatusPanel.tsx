import React from 'react'
import { transparentize } from 'polished'
import styled, { breakpoints, theme } from 'theme'

import { IProps, IState } from './StatusPanel.d'

const StatusPanel = styled.div`
  display: flex;
  flex: 0 1 auto;
  flex-flow: column nowrap;
  align-items: flex-start;
  justify-content: flex-end;
  border-top: 1px solid ${transparentize(0.75, theme.primaryColourDecorate)};
  padding-top: 0.5rem;
  margin-top: 0.5rem;

  @media ${breakpoints.desktop} {
    flex: 0 0 8vh;
    padding: 0;
    flex-flow: row nowrap;
    align-items: center;
  }
`

const TimeOfDay = styled.div`
  margin: 1rem 0 0 0;

  @media ${breakpoints.desktop} {
    margin: 0 0 0 auto;
  }
`

const PlayerStatus = styled.div`
  display: block;

  p {
    padding: 0;
    margin: 0;
  }
`

const Good = styled.span`
  color: ${theme.info};
  font-weight: bold;
`

const Bad = styled.span`
  color: ${theme.alert};
  font-weight: bold;
`

export default class extends React.PureComponent<IProps, IState> { // eslint-disable-line react/prefer-stateless-function
  // static defaultProps: Partial<IDefaultProps> = {}

  render() { // eslint-disable-line class-methods-use-this
    // const {} = this.props
    return (
      <StatusPanel>
        <PlayerStatus><p>You are <Bad>bleeding</Bad> and <Good>in good health</Good>.</p><p>You feel oppressed, and hungry.</p></PlayerStatus>
        <TimeOfDay>It is late in the day.</TimeOfDay>
      </StatusPanel>
    )
  }
}
