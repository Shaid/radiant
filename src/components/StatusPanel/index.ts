export { default } from './StatusPanel'

// NOTE: uncomment this line to pass through named exports
// export * from './StatusPanel'
