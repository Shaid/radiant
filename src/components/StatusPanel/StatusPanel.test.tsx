import React from 'react'
import { shallow } from 'enzyme'

import StatusPanel from '.'

describe('<StatusPanel />', () => {
  it('renders', () => {
    const wrapper = shallow(<StatusPanel />)
    expect(wrapper).toMatchSnapshot()
  })

  // @TODO: tests for StatusPanel
})
