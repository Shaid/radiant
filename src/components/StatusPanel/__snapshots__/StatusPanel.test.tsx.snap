// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusPanel /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusPanel__PlayerStatus>
          You feel oppressed, and hungry.
        </StatusPanel__PlayerStatus>,
        <StatusPanel__TimeOfDay>
          It is late in the day.
        </StatusPanel__TimeOfDay>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "You feel oppressed, and hungry.",
        },
        "ref": null,
        "rendered": "You feel oppressed, and hungry.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "It is late in the day.",
        },
        "ref": null,
        "rendered": "It is late in the day.",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusPanel__PlayerStatus>
            You feel oppressed, and hungry.
          </StatusPanel__PlayerStatus>,
          <StatusPanel__TimeOfDay>
            It is late in the day.
          </StatusPanel__TimeOfDay>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "You feel oppressed, and hungry.",
          },
          "ref": null,
          "rendered": "You feel oppressed, and hungry.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "It is late in the day.",
          },
          "ref": null,
          "rendered": "It is late in the day.",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
